### PR TITLE
Fix optimizely bug

### PR DESCRIPTION
**Pull request by vicky-g**
### Description

**Context:** As of right now, we use userIds to bucket users for certain features in Optimizely. Since users are initialized with userIds as `ANONYMOUS_USER`, if this id is deterministically bucketed by Optimizely to have a certain feature on, then effectively all users will have the feature on. 

**Problem:** This prevents proper feature rollout. 

**Solution:** 
1. Enable session based feature rollouts using a local storage Optimizely key as the unique identifier for users. 
2. Keep userId feature rollouts, but if `ANONYMOUS_USER`, disable

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.

1. Set feature flag to 100% in development mode (ex: `USE_TRACK_CONTENT_POLLING`) and set it to use the sessionId
2. Check to see that the feature is turned on
```
in audiusLibs:
useTrackContentPolling: true
```

1. Set feature flag to 100% in development mode (ex: `USE_TRACK_CONTENT_POLLING`) and set it to use the userId
2. Check to see if a feature flag is used and
  - userId is still `ANONYMOUS_USER`  --> feature is disabled
  -  userId is set --> feature is enabled

```
let flagEnabled = getFeatureEnabled(
    FeatureFlags.VICKY_DUMMY_FEATURE
  )

  if (flagEnabled) {
    console.log('1 i am enabled cus theres an id') 
  } else {
    console.log('1 i am not enabled cus theres no userid') // gets logged in console
  }

  // Set account ID in remote-config provider
  setUserId(account.user_id)

  flagEnabled = getFeatureEnabled(
    FeatureFlags.VICKY_DUMMY_FEATURE
  )

  if (flagEnabled) {
    console.log('2 i am enabled cus theres an id') 
  } else {
    console.log('2 i am not enabled cus theres no userid') // gets logged in console
  }
```

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/vg-fix-optimizely-bug

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/vg-fix-optimizely-bug

File: src/services/remote-config/FeatureFlags.ts [hunk 1]
```diff
@@ -7,10 +7,38 @@ export enum FeatureFlags {
...
+   */
+  USER_ID = 'user_id',
+  /**
+   * Segments feature experiments by a random uuid set in local storage defined by FEATURE_FLAG_SESSION_ID.
+   * There should always be a value for sessionId. This is managed in Provider.ts
+   */
+  SESSION_ID = 'session_id'
+}
+
+export const flagCohortType: {
```
Comment (vicky-g) [inline src/services/remote-config/FeatureFlags.ts:34]: todo: potentially follow up with sessionId fallback in another PR

**Review by raymondjacobson — APPROVED** [review]
LGTM! i really like this solution.
I think we'll eventually want some additional ways to slice expts, but this unblocks us right now for listens, etc.

**Comment by raymondjacobson** [comment]
curious for @rickyrombo  & @piazzatron  input

**Comment by raymondjacobson** [comment]
https://linear.app/audius/issue/AUD-621/rework-anonymous-ids-device-ids-in-feature-flag-system

**Review by piazzatron — APPROVED** [review]
Looks great! This is a nice solution and code is clean 👍

**Review by rickyrombo — APPROVED** [review]

**Pull request merged**